ocf/ocf.go

```diff
@@ -231,7 +241,11 @@ func NewEncoder(s string, w io.Writer, opts ...EncoderFunc) (*Encoder, error) {
 		Magic: magicBytes,
 		Meta:  cfg.Metadata,
 	}
-	_, _ = rand.Read(header.Sync[:])
+	if cfg.SyncSet {
+		header.Sync = cfg.Sync
+	} else {
```

Comment:
`else` is a code smell.

Reply:
Could you please elaborate? The behavior I wanted was to use the caller-supplied sync if one was indeed supplied, or (in the else branch) generate a random one.

Reply:
The `else` is not needed for this, and by reordering you can remove it, which is commonly the case with `else` statements, hence is it is a code smell.

```
	header.Sync = cfg.Sync
	if header.Sync == [16]byte{} {
		_, _ = rand.Read(header.Sync[:])
	}
```
The above does the same thing without the need for the `else`.

Reply:
I hadn't done this because I wanted to leave the possibility of one setting an all-zero sync block. But that's clearly not what the spec recommends, so I've changed the code as you've suggested.